MFCSample/MicroserviceManager/ServiceClient.cpp

```diff
@@ -0,0 +1,140 @@
+#include "pch.h"
+#include "ServiceClient.h"
+
+const string ServiceClient::SERVICE_REQUEST_EXCHANGE = "services_request";
+
+/**
+ * @brief 
+ * 
+ * @return 
+*/
+string ServiceClient::GenerateUUID()
+{
+    // Use a random device and a uniform distribution to generate random bytes
+    std::random_device rd;
+    std::mt19937 gen(rd());
+    std::uniform_int_distribution<> dis(0, 255);
+
+    // Generate 16 random bytes (UUID size)
+    std::vector<uint8_t> bytes(16);
+    for (int i = 0; i < 16; ++i) {
+        bytes[i] = static_cast<uint8_t>(dis(gen));
+    }
+
+    // Set the version (4) and variant (2) bits as required by UUID specification
+    bytes[6] = (bytes[6] & 0x0F) | 0x40; // version 4
+    bytes[8] = (bytes[8] & 0x3F) | 0x80; // variant 2
+
+    // Format the bytes as a string in the standard UUID format
+    std::stringstream ss;
+    ss << std::hex << std::uppercase << std::setfill('0');
+    for (const auto& byte : bytes) {
+        ss << std::setw(2) << static_cast<int>(byte);
+    }
+
+    return ss.str();
+}
+
+int ServiceClient::RequestService(std::string jsonData, std::function<void(const std::string&)> callback) {
+    amqp_connection_state_t conn = amqp_new_connection();
+    amqp_socket_t* socket = amqp_tcp_socket_new(conn);
```

Comment:
Hi Cuong,
for each Request a new Socket might be a performance issue?
Is it possible to use kind of socketpool here?
Thank you,
 Thomas

Reply:
Hi Thomas,
It's possible. I've just make the simple way for a running demo.
Will change in next update after some fixed bugs.
Thanks,
Cuong